services/account_type/master_payer.yaml

```diff
@@ -54,9 +71,161 @@ Resources:
+                - 'logs:PutLogEvents'
+              Resource: '*'
+            - Sid: AllowCreateCur
+              Effect: Allow
+              Action:
+                - 'cur:*'
+              Resource: '*'
+            - Sid: AllowVerifyCurBucket
+              Effect: Allow
+              Action:
+                - 's3:*'
+              Resource:
+                - !Sub
+                  - 'arn:aws:s3:::${BucketName}-${Id}'
+                  - { BucketName: !FindInMap [ Defaults, Cur, BucketName ],
+                      Id: !Select [ 2, !Split [ '/', !Ref 'AWS::StackId' ] ] }
+                - !Sub
+                  - 'arn:aws:s3:::${BucketName}-${Id}/*'
+                  - { BucketName: !FindInMap [ Defaults, Cur, BucketName ],
+                      Id: !Select [ 2, !Split [ '/', !Ref 'AWS::StackId' ] ] }
+
+  # NOTE: This is an inline lambda function b/c we want it to be easy for you
+  # to see the resource we're creating, in this case a Cost and Usage report.
+  # We always prefer pure CFN to custom resources; however, CFN currently does
+  # not support creating a CUR.
+  CostAndUsageReportLambda:
+    Condition: CreateCurAndBucketAndPolicy
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Description: Create/Delete Cost and Usage Report
+      Handler: index.handler
+      Runtime: python3.7
+      Role: !GetAtt CostAndUsageReportRole.Arn
+      Timeout: 15
+      # Unfortunately, the default lambda python execution environment includes an older
+      # version of boto3 that does not support the ReportDefinition parameters we need.
+      # This layer contains a newer version of boto3.
+      Layers:
+        - !Sub 'arn:aws:lambda:${AWS::Region}:113088814899:layer:Klayers-python37-boto3:7'
+      Code:
+        ZipFile: |
+          import boto3
+          import cfnresponse
+          import json
+          import logging
+
+          cur = boto3.client('cur', region_name='us-east-1')  # cur is only in us-east-1
+          logger = logging.getLogger()
+          logger.setLevel(logging.INFO)
+
+
+          def handler(event, context):
+              result = cfnresponse.SUCCESS
+              report_name = 'cloudzero-cur-hourly-csv'
+              try:
+                  logger.info(f'Received event: {event}')
+                  bucket_name = event['ResourceProperties']['BucketName']
+                  current_region = event['ResourceProperties']['Region']
+                  valid_regions = {'us-west-1', 'us-west-2', 'ap-northeast-1', 'ap-northeast-3', 'eu-north-1', 'ap-southeast-1',
+                                   'eu-west-1', 'us-east-1', 'ap-southeast-2', 'eu-central-1', 'ap-east-1'}
+                  region = current_region if current_region in valid_regions else 'us-east-1'
+                  s3 = boto3.resource('s3', region_name=region)
+                  bucket = s3.Bucket(bucket_name)
+
+                  if event['RequestType'] == 'Create':
+                      bucket.create()
```

Comment:
This makes me so sad =(:
- we cannot use pure CFN for even the Bucket and Bucket Policy b/c the CUR does not support the S3Bucket in _all_ regions, only the regions listed above
- so if we want to _ensure_ we can always create a valid CUR, we need to default to a known region that CUR supports, in this case, `us-east-1`
- I really don't like this for a bunch of reasons, not least of which is that this resource may create a CUR and S3 Bucket in some another region ... but it works